### PR TITLE
Docs: DO-2316: updating layout docs

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Motivation and Context

Some people had difficulties with hug and Graph/Table components. Based on feedback it was worth adding a small section in the docs briefly explaining the issue.

## Implementation Description

Updated the docs on lay it out page to include explanation of why some components became so small within a hug component.

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [ ] I have added relevant tests (unit, integration or regression).
- [ ] I have added comments to all the bits that are hard to follow.
- [x] I have added/updated Documentation.
- [ ] I have updated the appropriate changelog with a line for my changes.

### CODE REVIEW

**Review by tamara-causalens — APPROVED** [review]
Great!

**Review by krzysztof-causalens — APPROVED** [review]

File: docs/best-practices/laying-out-apps.md [hunk 1]
```diff
@@ -161,6 +161,14 @@ Stack(
 
 ```
 
+:::caution
+
+**When dealing with `CausalGraphViewer`, `Table`, `Plotly` and `Scrollable Stacks`:**
+These components are special in that they essentially "scan" the space available on the page before they render. This is good because it allows them to adapt to their surroundings but it comes with drawbacks.
+
+If you surround them with hug, hug tries to make the space only as big as it needs to. So there is a conflict of interest as there is no space leftover for them to take. In that case they will take their default `min_height`. So if you are using these components within a `hug` container you may need to overwrite `min_height` of the component should you want them to take more space on the screen.
```
Comment (claudiawestby) [inline docs/best-practices/laying-out-apps.md:169]: "If you surround them with hug" this sounds a bit out of context. Can we be more explicit ie "If you place them in a container that has `hug` set to `True`"

File: docs/best-practices/laying-out-apps.md [hunk 2]
```diff
@@ -161,6 +161,14 @@ Stack(
 
 ```
 
+:::caution
+
+**When dealing with `CausalGraphViewer`, `Table`, `Plotly` and `Scrollable Stacks`:**
```
Comment (claudiawestby) [inline docs/best-practices/laying-out-apps.md:166]: If we put scrollable stacks in capital letters and within `code brackets` it seems like it is its own component which could send someone looking for it. could we do scrollable `Stack`s?

File: docs/best-practices/laying-out-apps.md (hunk 1 above)
Comment (patricia-causalens) [reply docs/best-practices/laying-out-apps.md]: Good point, addressed

File: docs/best-practices/laying-out-apps.md (hunk 2 above)
Comment (patricia-causalens) [reply docs/best-practices/laying-out-apps.md]: fair point, fixed

**Pull request merged**